ann-mukundi/questioner
# #164297869 Configure PostgreSQL as default DB Engine

#### What does this PR do?
Configures PostgreSQL as the default Database Engine
#### Description of Task to be completed?
- Change the database configuration default in the settings.py file to PostgreSQL with the database url as environmental variables 
- Update the .env-example file to include the database url variables
#### How should this be manually tested?
1. git clone repo or git pull origin develop
2. If you had ran migrations, you will need to grant some privileges to make sure the permissions necessary for the relation django_migrations exist. Do this by running the following commands in the terminal:
- sudo su postgres
- psql drf -c "GRANT ALL ON ALL TABLES IN SCHEMA public to django;"
- psql drf -c "GRANT ALL ON ALL SEQUENCES IN SCHEMA public to django;" 
- psql drf -c "GRANT ALL ON ALL FUNCTIONS IN SCHEMA public to django;"
- exit
- python api/manage.py runserver
There should be no **django.db.utils.ProgrammingError** now
#### Any background context you want to provide?
PostgreSQL works based on a client-server model which requires a DB server to set up and run over the network. It is better than the default SQLite database provided by Django as PostgreSQL gives you the functionality to maintain such a complex database smoothly without limitations and is also recommended data integrity and reliability is highly concerned.
#### What are the relevant pivotal tracker stories?
[#164297869](https://www.pivotaltracker.com/n/projects/2315729)
#### Screenshots (if appropriate)
The app should run to view this page
![screenshot_20190305_175408](https://user-images.githubusercontent.com/41931013/53813951-c6c17e00-3f6f-11e9-9c53-c6ec8f13f0f9.png)
